sql-plugin/src/main/scala/com/nvidia/spark/rapids/spill/SpillFramework.scala

```diff
@@ -318,16 +333,32 @@ class SpillableHostBufferHandle private (
     materialized
   }
 
+  private var toSpill: Option[HostMemoryBuffer] = None
```

Comment:
Can we have some comments about how toSpill is used with synchronization? Just so it is clear that this is used to gate when a spill is actively being done.